FlexMeasures/flexmeasures
# Status page: show by source type

## Description

1. Split status by data source type
2. Change staleness calculation for Forecaster / Scheduler sources (staleness should be calculated as difference between now and max event start time)

## Look & Feel

1. "/<id>/status/" page should return source key as well
2. UI status page should include source type

![asset-status-source-type](https://github.com/user-attachments/assets/003576ae-4692-4e0c-85c5-0cd0b6996c4e)

![Screenshot from 2025-03-17 17-18-33](https://github.com/user-attachments/assets/816df739-8f28-4fa5-bed5-fdbe46ec4339)


## How to test

Is tested in test_sensor_data_schema.py (test_get_status_multi_source)
Can be tested the following way - add data from several sources to some sensor. Go to its status page and check that statuses are split by this source column + scheduler/forecaster statuses are calculated differently 

## Further Improvements

-

## Related Items

Follow up for https://github.com/FlexMeasures/flexmeasures/pull/941

---
